src/coreclr/jit/llvm.cpp

```diff
@@ -1656,25 +1685,32 @@ void Llvm::storeLocalVar(GenTreeLclVar* lclVar)
 {
     if (lclVar->gtFlags & GTF_VAR_DEF)
     {
-        Value* valueRef = getGenTreeValue(lclVar->gtGetOp1());
-        assert(valueRef != nullptr);
+        Value* localValue = getGenTreeValue(lclVar->gtGetOp1());
+        assert(localValue != nullptr);
+
+        unsigned lclNum = lclVar->GetLclNum();
+        Value* allocaValue = (*m_allocas)[lclNum];
+        if (allocaValue != nullptr)
+        {
+            _builder.CreateStore(localValue, castIfNecessary(allocaValue, localValue->getType()->getPointerTo()));
+        }
+
         // This could be done in the NE operator, but sometimes that would be needless, e.g. when followed by JTRUE
         // TODO-LLVM: As this is a zero extend widening operation, this is only valid if the small int is unsigned.  We don't know that here, so likely it would be better to
         // delete this and do the cast in the operator.  It seems likely that the cast will be a nop anyway, at least in Wasm, as Wasm does not have any number types smaller than i32
-        if (valueRef->getType()->isIntegerTy())
+        if (localValue->getType()->isIntegerTy())
         {
-            valueRef = zextIntIfNecessary(valueRef);
+            localValue = zextIntIfNecessary(localValue);
         }
 
         LclVarDsc* varDsc = _compiler->lvaGetDesc(lclVar);
-
-        if (varDsc->lvIsParam)
+        if (varDsc->lvIsParam) //TODO-LLVM: not doing params yet
         {
             failFunctionCompilation();
         }
 
-        SsaPair ssaPair = {lclVar->GetLclNum(), lclVar->GetSsaNum()};
-        _localsMap->insert({ssaPair, valueRef });
+        SsaPair ssaPair = {lclNum, lclVar->GetSsaNum()};
```

Comment:
Will it store `RESERVED_SSA_NAME` for locals live on the frame?

That doesn't seem right, there should probably be an early return above, or an `if-else` structure (which could also be said to be better).

Reply:
Thanks, changed to if/else